app/src/main/kotlin/com/wire/android/ui/home/HomeScreen.kt

```diff
@@ -34,6 +38,13 @@ fun HomeScreen(startScreen: String?, viewModel: HomeViewModel, syncViewModel: Sy
     viewModel.checkRequirements()
     val homeState = rememberHomeState()
 
+    val snackbarHostState = remember { SnackbarHostState() }
+    handleSnackBarMessage(snackbarHostState, viewModel.snackBarMessageState)
+
+    LaunchedEffect(viewModel.savedStateHandle) {
+        viewModel.showPendingToast()
```

Comment:
I think this should not be the desired flow. The view should never tell the VM what to do. Instead, it should just listen/observe to events being updated on the VM and then react on them

Reply:
I think we should refactor this (and maybe also the logic handling the back navigation args mapping inside `ConversationScreen`, so that we collect any change on the VM appropriate vars and react on them. Something similar to what we do inside the `SelfUserProfileScreen` with the VM `errorMessageCode`:

```
state.errorMessageCode?.let { errorCode ->
        val errorMessage = mapErrorCodeToString(errorCode)
        LaunchedEffect(errorMessage) {
            snackbarHostState.showSnackbar(errorMessage)
            onMessageShown()
        }
    }

Reply:
Will try to suggest another approach 💪